fluidsynth/src/synth/fluid_synth.c

```diff
@@ -2501,6 +2559,25 @@ fluid_synth_nwrite_float(fluid_synth_t* synth, int len,
 #endif //WITH_FLOAT
     }
 
+    for (i = 0; i < synth->effects_channels; i++)
+    {
+#ifdef WITH_FLOAT
+      if(fx_left != NULL)
+        FLUID_MEMCPY(fx_left[i + count], fx_left_in[i], bytes);
+      
+      if(fx_right != NULL)
+        FLUID_MEMCPY(fx_right[i + count], fx_right_in[i], bytes);
+#else //WITH_FLOAT
+      int j;
+      for (j = 0; (fx_left != NULL) && (j < num); j++)
+          fx_left[i][j + count] = (float) fx_left_in[i][j];
```

Comment:
Oh ok, I usually trust todays compilers to generate something useful, but older compilers might have problems here, I agree.